packages/@aws-cdk/aws-ecs/lib/ecs/ecs-cluster.ts

```diff
@@ -0,0 +1,227 @@
+import autoscaling = require('@aws-cdk/aws-autoscaling');
+import cloudwatch = require ('@aws-cdk/aws-cloudwatch');
+import ec2 = require('@aws-cdk/aws-ec2');
+import iam = require('@aws-cdk/aws-iam');
+import cdk = require('@aws-cdk/cdk');
+import { BaseCluster, BaseClusterProps } from '../base/base-cluster';
+
+/**
+ * Properties to define an ECS cluster
+ */
+export interface EcsClusterProps extends BaseClusterProps {
+  /**
+   * Whether or not the containers can access the instance role
+   *
+   * @default false
+   */
+  containersAccessInstanceRole?: boolean;
+
+  /**
+   * The type of EC2 instance to launch into your Autoscaling Group
+   */
+  instanceType?: ec2.InstanceType;
+
+  /**
+   * Number of container instances registered in your ECS Cluster
+   *
+   * @default 1
+   */
+  size?: number;
+}
+
+/**
+ * A container cluster that runs on your EC2 instances
+ */
+export class EcsCluster extends BaseCluster implements IEcsCluster {
+  /**
+   * Import an existing cluster
+   */
+  public static import(parent: cdk.Construct, name: string, props: ImportedEcsClusterProps): IEcsCluster {
+    return new ImportedEcsCluster(parent, name, props);
+  }
+
+  /**
+   * The AutoScalingGroup that the cluster is running on
+   */
+  public readonly autoScalingGroup: autoscaling.AutoScalingGroup;
+
+  /**
+   * SecurityGroup of the EC2 instances
+   */
+  public readonly securityGroup: ec2.SecurityGroupRef;
+
+  constructor(parent: cdk.Construct, name: string, props: EcsClusterProps) {
+    super(parent, name, props);
+
+    const autoScalingGroup = new autoscaling.AutoScalingGroup(this, 'AutoScalingGroup', {
+      vpc: props.vpc,
+      instanceType: props.instanceType || new ec2.InstanceTypePair(ec2.InstanceClass.T2, ec2.InstanceSize.Micro),
+      machineImage: new EcsOptimizedAmi(),
+      updateType: autoscaling.UpdateType.ReplacingUpdate,
+      minSize: 0,
+      maxSize: props.size || 1,
+      desiredCapacity: props.size || 1
+    });
+
+    this.securityGroup = autoScalingGroup.connections.securityGroup!;
+
+    // Tie instances to cluster
+    autoScalingGroup.addUserData(`echo ECS_CLUSTER=${this.clusterName} >> /etc/ecs/ecs.config`);
+
+    if (!props.containersAccessInstanceRole) {
+      // Deny containers access to instance metadata service
+      // Source: https://docs.aws.amazon.com/AmazonECS/latest/developerguide/instance_IAM_role.html
+      autoScalingGroup.addUserData('sudo iptables --insert FORWARD 1 --in-interface docker+ --destination 169.254.169.254/32 --jump DROP');
```

Comment:
For awsvpc mode (which doesn't use docker0), you should also set ECS_AWSVPC_BLOCK_IMDS=true
https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task-iam-roles.html